pkgs/development/compilers/rust/1_82.nix

```diff
@@ -105,7 +105,13 @@ import ./default.nix
 
     # Note: the version MUST be one version prior to the version we're
     # building
-    bootstrapVersion = "1.81.0";
+    bootstrapVersion =
+      # Exceptionally we build the riscv64-linux compiler from the same
+      # bootstrap version when compiling natively because the distributed
+      # 1.81.0 binary cannot build the 1.82.0 compiler on that platform.
+      # https://github.com/rust-lang/rust/issues/129268.
+      # Remove this when bumping to 1.83.0.
+      if stdenv.targetPlatform.isRiscV64 then "1.82.0" else "1.81.0";
```

Comment:
Shouldn't this be buildPlatform?  Whether the 1.82.0 tarball is used depends on whether we're building _from_ RISC-V, right?

But also, maybe it would be better to just use 1.82.0 on all platforms for this release, to avoid confusing divergences like this?  I'm pretty sure the comment above that says "MUST be one version prior" is not accurate, because my understanding is that rustc also always supports building a released compiler with itself.